### PR TITLE
feat(analytics): track MCP compatibility adoption

### LINKED ISSUE #139: feat(analytics): add MCP package and API compatibility adoption tracking
## Background

MCP adoption depends on clients staying compatible. We need to know which package versions are in use and help users upgrade when stale.

## Goal

Track MCP versions in use and whether clients are stale or incompatible with the live API.

## Current Behavior

Doctor/status can detect some local setup issues, but adoption and compatibility are not aggregated into product analytics.

## Desired Behavior

MCP clients report safe version metadata and the control panel/API can show stale or incompatible client distribution.

## Implementation Requirements

- Emit redacted client version/product events from CLI/MCP requests.
- Expose API compatibility metadata.
- Improve doctor/status upgrade hints where needed.
- Mock npm registry lookups in tests.
- Do not log auth tokens or local absolute paths.

## Public/Private Output Boundaries

Public GitHub output must never include wallets, hotkeys, payout/reward estimates, raw trust scores, public score estimates, private reviewability, private scoreability context, or farming language. Private API/MCP/control-panel surfaces may show authenticated scoreability, blockers, projections, and evidence, but must not claim guaranteed payout outcomes.

## Acceptance Criteria

- doctor/status can report stale installs.
- Control panel can show MCP version distribution.
- API exposes compatibility metadata.
- Tests mock npm registry and API version responses.

## Testing Requirements

- `npm run test:ci` must pass.
- Global coverage must remain at or above 97% for lines, statements, functions, and branches.
- Aim for 98%+ branch coverage locally.
- Add tests for every new branch, fallback path, sanitizer rule, and regression.
- Add invariant/property-style tests when behavior depends on sorting, gating, scoring, source-upload safety, public/private boundaries, or telemetry privacy.
- MCP/local tooling must prove source contents are not uploaded when local metadata is involved.
- Public GitHub output must be tested against forbidden language: wallet, hotkey, raw trust score, payout, reward estimate, farming, private reviewability, and public score estimate.

### PR DESCRIPTION
**Pull request by oktofeesh1**
## Summary

Closes #139.

Adds privacy-safe MCP compatibility adoption tracking across the CLI/MCP request path, private analytics APIs, and the operator dashboard.

## What changed

- Added bounded MCP package/client/protocol telemetry parsing with unsafe header rejection.
- Sent static MCP package/client/version headers from the CLI API client.
- Recorded MCP telemetry on MCP success and error paths, including compatibility status.
- Added aggregate MCP compatibility adoption summaries for private operator/API analytics.
- Surfaced stale/incompatible MCP client counts and version buckets in the analytics dashboard.
- Regenerated the app OpenAPI snapshot for the new private analytics endpoint.
- Added regression tests for no-leak invariants, endpoint clamps, summary aggregation, CLI headers, and MCP error telemetry.

## Why

MCP adoption needs version distribution and stale/incompatible client visibility without storing tokens, local paths, source contents, or raw actor/session identifiers in product analytics.

## Validation

- `npm run test:unit -- test/unit/mcp-compatibility.test.ts test/unit/product-usage-mcp-adoption.test.ts`
- `npm run test:unit -- test/unit/mcp-server-telemetry.test.ts`
- `npm run test:unit -- test/unit/mcp-cli.test.ts`
- `npm run test:integration -- test/integration/api.test.ts`
- `npm run test:ci`
- Codex Security diff scan: no reportable findings; every final-diff worklist row completed and the markdown/HTML report rendered locally.

## Notes

- This branch is stacked after #182 and #183; it should be reviewed after those analytics foundations land.
- Public output remains aggregate-only and does not expose tokens, local paths, raw actor names, raw session IDs, source contents, wallet/hotkey data, or private scoring context.

### CODE REVIEW

**Review by JSONbored — APPROVED** [review]
@oktofeesh1 This is ready from my side. The MCP compatibility adoption tracking is now green and merge-clean against current `main`.

A few notes:
- The API/OpenAPI/MCP surfaces are updated together.
- CLI telemetry, compatibility service behavior, MCP server telemetry, and adoption aggregation all have targeted tests.
- This should be merged after the rollup foundation if we are keeping the analytics stack ordered cleanly.

File: src/api/routes.ts [hunk 1]
```diff
@@ -713,12 +809,28 @@ export function createApp() {
         { label: "Installed coverage", value: installedRepos, spark: sparklineFromCounts(installedRepos, Math.max(repositories.length, 1)) },
       ],
       weeklyReport: buildOperatorWeeklyReport({ repositories, installations, health, registry, scoring, upstreamDrift }),
+      usageSummary,
+      usageRollups,
+      usageRollupStatus,
+      mcpCompatibilityAdoption,
       registry,
       scoringModel: scoring,
```
[the remark this answers is not in the document]
Comment (oktofeesh1) [reply src/api/routes.ts:817]: Current head has this covered: `src/api/routes.ts` guards both `/v1/app/analytics/mcp-compatibility` and `/v1/app/analytics/daily-rollups` with `requireAppRole(c, ["operator"])` before returning data. `test/integration/api.test.ts` also asserts non-operator sessions receive `403` for both analytics endpoints. Latest Superagent run on this head passed with `No suspicious PR changes were detected`; validation also passed with `npm run test:ci`.
